### PR TITLE
Faster Travis, Test Fixes and better error messages

### PR DESCRIPTION
**Pull request by jsalvatier**
I've split the tests into examples and the rest, though they don't actually seem to run in parallel. I've also tried to eliminate some of the requirements.

I've got the tests working again. The major issue was that metropolis called nested functions and Python can't deal with that in multiprocessing (can't pickle them).

### CODE REVIEW

**Comment by fonnesbeck** [comment]
Faster Travis would be a great band name.

File: .travis.yml [hunk 1]
```diff
@@ -2,16 +2,17 @@ language: python
 python:
   - "2.7"
 before_install:
-  - sudo apt-get install liblapack-dev gfortran python-scipy libblas-dev libsuitesparse-dev swig
```
Comment (twiecki) [inline .travis.yml:5]: These are required for building scipy.

**Comment by jsalvatier** [comment]
Damn, I guess it needs Cython too. 

@twiecki Does it also need to sudo apt-get python-scipy?

**Comment by twiecki** [comment]
No, unfortunately you can't install python modules via apt on travis. travis creates a virtualenv that only has access to local packages installed into the virtualenv, so you have to install everything via pip. These are just the dependencies for scipy.

**Comment by jsalvatier** [comment]
Can someone look over this and merge if its ready? I really want the tests working again.

**Pull request merged**